demo/android/OrcaDemo/orca-demo-app/src/main/java/ai/picovoice/orcademo/MainActivity.java

```diff
@@ -322,25 +350,199 @@ private void stopPlayback() {
+                });
+            } else {
+                orcaStream.close();
+                orcaStream = null;
+                runOnUiThread(() -> streamSecsTextView.setVisibility(View.INVISIBLE));
+            }
+        } catch (OrcaException e) {
+            onOrcaException(e);
+        }
+    }
+
     public void onSynthesizeClick(View view) {
         if (orca == null) {
             displayError("Orca is not initialized");
             synthesizeButton.setChecked(false);
             return;
         }
 
-        if (synthesizeButton.isChecked()) {
-            String text = synthesizeEditText.getText().toString();
-            if (!previousText.equals(text)) {
-                runSynthesis(text);
+        String text = synthesizeEditText.getText().toString();
+        if (orcaStream == null) {
+            if (synthesizeButton.isChecked()) {
+                if (!previousText.equals(text)) {
+                    runSynthesis(text);
+                } else {
+                    startPlayback();
+                }
             } else {
-                startPlayback();
+                stopPlayback();
             }
         } else {
-            stopPlayback();
+            runStreamSynthesis(text);
+        }
+    }
+
+    private void stopStreamPlay() {
+        try {
+            runOnUiThread(() -> {
+                setUIState(UIState.EDIT);
+                infoTextView.setText("");
+                streamTextView.setVisibility(View.INVISIBLE);
+                synthesizeEditText.setVisibility(View.VISIBLE);
+            });
+        } catch (Exception e) {
+            displayError(e.toString());
         }
     }
 
+    private void runStreamSynthesis(final String text) {
+        runOnUiThread(() -> {
+            setUIState(UIState.PLAYBACK);
+            infoTextView.setText("Streaming...");
+            streamTextView.setVisibility(View.VISIBLE);
+            streamSecsTextView.setVisibility(View.VISIBLE);
+            synthesizeEditText.setVisibility(View.INVISIBLE);
+        });
+
+        AtomicBoolean isStreamingText = new AtomicBoolean(false);
+        ArrayList<String> textStream = new ArrayList<>();
+
+        AtomicBoolean isQueueingStreamingPcm = new AtomicBoolean(false);
+        ConcurrentLinkedQueue<short[]> pcmQueue = new ConcurrentLinkedQueue<>();
+        CountDownLatch latch = new CountDownLatch(1);
+
+        executor.submit(() -> {
+            isStreamingText.set(true);
+
+            String[] words = text.split(" ");
+            for (String word : words) {
+                word += " ";
+                try {
+                    String finalWord = word;
+                    mainHandler.post(() -> {
+                        textStream.add(finalWord);
+                        streamTextView.append(finalWord);
+                    });
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
```

Comment:
this won't report the error btw, it'll just crash the thread. If you want the app to do something on this error, you need to report back via a change to the ui or something